get-deployment-metrics.py

```diff
@@ -0,0 +1,286 @@
+        "--detailed", help="Show detailed output for each repo", action="store_true"
+    )
+    parser.add_argument(
+        "--include-manual-runs",
+        help="Include manual workflow runs in stats computations",
+        dest="include_manual_runs",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--verbose", help="Turn on DEBUG logging", action="store_true", required=False
+    )
+
+    args = parser.parse_args()
+
+    log_level = logging.INFO
+
+    if args.verbose:
+        print("Verbose logging selected")
+        log_level = logging.DEBUG
+
+    # Setup some logging
+    logger = logging.getLogger()
+    logger.setLevel(log_level)
+    ch = logging.StreamHandler()
+    ch.setLevel(log_level)
+    console_formatter = logging.Formatter("%(levelname)8s: %(message)s")
+    ch.setFormatter(console_formatter)
+    logger.addHandler(ch)
+
+    # Initialize connection to Github API
+    github_handle = GitHub(token=args.github_pat, paginate=True)
+    # github_handle = GitHub(token=args.github_pat)
+
+    # Get all the repos in the org
+    # /orgs/{org}/repos
+    gh_status, repo_data = github_handle.orgs[args.org_name].repos.get()
+
+    for repo in repo_data:
+        repo_name = repo["name"]
+        repo_printed = False
+
+        logger.debug("Processing repo {}".format(repo_name))
+
+        if repo["archived"]:
+            logger.debug("repo {} is archived - skipping".format(repo_name))
+            continue
+
+        # Now for each repo, see if we have a deployment workflow matching the pattern
+        # /repos/{org}/{repo name}/actions/workflows
+        gh_status, workflow_data = github_handle.repos[args.org_name][
+            repo_name
+        ].actions.workflows.get()
+
+        for workflow in workflow_data["workflows"]:
+            workflow_success_count = 0
+            workflow_success_rate = 100
+
+            workflow_fail_count = 0
+            workflow_failure_rate = 0
+
+            workflow_avg_duration = 0
+            workflow_total_duration = 0
+
+            workflow_id = workflow["id"]
+            workflow_name = workflow["name"]
+            workflow_summary_name = workflow_name.replace(
+                " ", ""
+            )  # Dicts cannot have spaces in keys
+
+            logging.debug("Found workflow {}".format(workflow_name))
+
+            if fnmatch.fnmatch(workflow_name, args.workflow_pattern):
+                logging.debug(
+                    "workflow {} matches {}".format(
+                        workflow_name, args.workflow_pattern
+                    )
+                )
+
+                # We have a matching workflow - get the runs for it in our timeframe
+                # repos/{org}}/{repo name}/actions/workflows/{workflow id}/runs
+                gh_status, workflow_runs = (
+                    github_handle.repos[args.org_name][repo_name]
+                    .actions.workflows[workflow_id]
+                    .runs.get(created=args.date_filter)
```

Comment:
FC: Should we add some timestamp syntax validation to `args.date_filter` since this is coming directly from the user?

Reply:
The [github date filter](https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates) is pretty broad and I can't find a native validator so I'm not going to validate it.  However, I ran some testing with an invalid date and got back an error so I will handle this